testRearrangeCarGroup8.py

```diff
@@ -0,0 +1,118 @@
+import unittest
+import numpy as np
+from RearrangeCarGroup8 import RearrangeCars
+
+class TestRearrangeCars(unittest.TestCase):
+
+    '''Unittests for RearrangeCars class and promised fuctionality
+    '''
+
+    def setUp(self):
+
+        '''Basic setting that will be used in several tests
+        '''
+        
+        self.initial_car_order = [1, 2, 0, 3]
+        self.final_car_order = [3, 1, 2, 0]
+        self.rc = RearrangeCars(list(self.initial_car_order), list(self.final_car_order))

```

Comment:
@luxiya01 Now I got your explanation, many thanks. Basically indeed we need the line self.current_car_order = list(self.initial_car_order) in _variables_to_init_state(self). My tests also confirmed that. 

```
def test_initial_order_preserved(self):

        '''Tests whether, after having called rearrange_all_cars() once,
        the field self.initial_car_order is still unchanged

        Note: This test must pass for test_rearrange_sample_cars_twice to
        be meaningful
        '''
        
        print 'test for changing the var'
        
        self.initial_car_order = [3, 1, 2, 0]
        self.rc.rearrange_all_cars(print_console=True)
        print 'changed car order', self.rc.initial_car_order
        self.assertEqual(self.rc.initial_car_order, self.initial_car_order)
```